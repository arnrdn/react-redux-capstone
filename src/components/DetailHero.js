import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faChevronDown, faChevronUp } from '@fortawesome/free-solid-svg-icons';
import img from '../img/details.svg';

const DetailHero = ({
  ticker, name, price, changes,
}) => (
  <div className="hero-detail">
    <div className="hero-content-detail">
      <div className="info">
        <div className="detail-name">
          <h1>{name}</h1>
          <span>{ticker}</span>
        </div>
        <div className="detail-hero-finance">
          <div className="price">
            <b>$</b>
            {' '}
            {price}
          </div>
          <div className="changes">
            {changes > 0 ? (<FontAwesomeIcon className="arrow" icon={faChevronUp} color="green" />)
              : (<FontAwesomeIcon className="arrow" icon={faChevronDown} color="red" />)}
            {changes}
          </div>
        </div>
      </div>
      <img
        className="hero-img"
        src={img}
        alt="person looking at graphs"
      />
    </div>
  </div>
);

DetailHero.propTypes = {
  ticker: PropTypes.string,
  name: PropTypes.string,
  price: PropTypes.number,
  changes: PropTypes.number,
}.isRequired;

export default DetailHero;
