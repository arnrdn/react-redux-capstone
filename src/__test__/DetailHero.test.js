import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import DetailHero from '../components/DetailHero';

describe('App component should render as expected', () => {
  test('render Details', () => {
    const details = render(
      <Provider store={store}>
        <BrowserRouter>
          <DetailHero
            name="Ford Motor Company"
            ticker="F"
            price="18.08"
            changes={0.620001}
          />
        </BrowserRouter>
      </Provider>,
    );
    expect(details).toMatchSnapshot();
  });
});
