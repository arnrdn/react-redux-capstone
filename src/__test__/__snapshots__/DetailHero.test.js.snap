// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render as expected render Details 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="hero-detail"
      >
        <div
          class="hero-content-detail"
        >
          <div
            class="info"
          >
            <div
              class="detail-name"
            >
              <h1>
                Ford Motor Company
              </h1>
              <span>
                F
              </span>
            </div>
            <div
              class="detail-hero-finance"
            >
              <div
                class="price"
              >
                <b>
                  $
                </b>
                 
                18.08
              </div>
              <div
                class="changes"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-chevron-up arrow"
                  color="green"
                  data-icon="chevron-up"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M416 352c-8.188 0-16.38-3.125-22.62-9.375L224 173.3l-169.4 169.4c-12.5 12.5-32.75 12.5-45.25 0s-12.5-32.75 0-45.25l192-192c12.5-12.5 32.75-12.5 45.25 0l192 192c12.5 12.5 12.5 32.75 0 45.25C432.4 348.9 424.2 352 416 352z"
                    fill="currentColor"
                  />
                </svg>
                0.620001
              </div>
            </div>
          </div>
          <img
            alt="person looking at graphs"
            class="hero-img"
            src="details.svg"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="hero-detail"
    >
      <div
        class="hero-content-detail"
      >
        <div
          class="info"
        >
          <div
            class="detail-name"
          >
            <h1>
              Ford Motor Company
            </h1>
            <span>
              F
            </span>
          </div>
          <div
            class="detail-hero-finance"
          >
            <div
              class="price"
            >
              <b>
                $
              </b>
               
              18.08
            </div>
            <div
              class="changes"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-up arrow"
                color="green"
                data-icon="chevron-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 352c-8.188 0-16.38-3.125-22.62-9.375L224 173.3l-169.4 169.4c-12.5 12.5-32.75 12.5-45.25 0s-12.5-32.75 0-45.25l192-192c12.5-12.5 32.75-12.5 45.25 0l192 192c12.5 12.5 12.5 32.75 0 45.25C432.4 348.9 424.2 352 416 352z"
                  fill="currentColor"
                />
              </svg>
              0.620001
            </div>
          </div>
        </div>
        <img
          alt="person looking at graphs"
          class="hero-img"
          src="details.svg"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
